domain-concept/src/main/java/life/qbic/domain/concepts/DomainEventDispatcher.java

```diff
@@ -0,0 +1,46 @@
+package life.qbic.domain.concepts;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * <b>Domain Event Dispatcher</b>
+ * <p>
+ * Dispatches domain events to registered {@link DomainEventSubscriber}.
+ *
+ * <p>
+ * <strong>Disclaimer</strong>
+ * <p>The implementation runs in the main application thread and is blocking. Depending on the
+ * number of registered subscriber and their implementation, expect the dispatching of events to
+ * block your main app.</p>
+ *
+ * @since 1.0.0
+ */
+public class DomainEventDispatcher {
+
+  private static DomainEventDispatcher INSTANCE;
+  private final List<DomainEventSubscriber<? extends DomainEvent>> subscribers;
```

Comment:
I think DomainEventSubscriber should have the bound in the generic type. You can call this with instead.
```suggestion
  private final List<DomainEventSubscriber<?>> subscribers;
```